### DISCUSSION TITLE
Not receiving LoRa message

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#22455] [Q&A] [answered]
I has two of [Lilygo T3_V1.61](https://www.lilygo.cc/products/lora32-v2-1_1-6-us?pr_prod_strat=e5_desc&pr_rec_id=1f1853463&pr_rec_pid=7792345546933&pr_ref_pid=7286884008117&pr_seq=uniform) (433MHz).

I compiled with:
```
#define USE_SPI_LORA            // Add support for LoRaSend and LoRaCommand (+4k code)
#define USE_LORA_SX127X         // Add driver support for LoRa on SX127x based devices like M5Stack LoRa868, RFM95W (+5k code)
#define USE_LORAWAN_BRIDGE      // Not needed yet
```

both boards found lora module:
` LOR: SX127x initialized`

are configed with:
`04:57:55.661 RSL: RESULT = {"LoRaConfig":{"Frequency":868.0,"Bandwidth":125.0,"SpreadingFactor":9,"CodingRate4":7,"SyncWord":18,"OutputPower":10,"PreambleLength":8,"CurrentLimit":60.0,"ImplicitHeader":0,"CrcBytes":2}}`

`lorasend "test"` seems to work but nothing is received.
i found no information in [doku ](https://tasmota.github.io/docs/LoRa-and-LoRaWan-Bridge/#lora-commands) or source code. Sorry

Which steps have I skipped?
Is there a node address that can be set and addressed?

**Comment by arendst** [comment]
You say have 433 Mhz and configured 868 Mhz. This is likely to fail.

**Reply by sfromis** [reply]
That specific board works for me, when I follow the docs about configuring the frequency.

Not important, but don't wrap the string in `"`, unless you want those to be part of the message.

**Reply by ChrisVonFrosta** [reply]
> You say have 433 Mhz and configured 868 Mhz. This is likely to fail.

i thought it not matter if 433 or 868 is set, because i though the different should be only the antenna matching and since both devices are close on the table,i thought it should work.
Anyway, for sure, i tried also with LoRaConfig {"Frequency":433.0} on both sides.
No effects.

Board 1:
```
10:24:05.787 RSL: RESULT = {"LoRaConfig":{"Frequency":433.0,"Bandwidth":125,"SpreadingFactor":9,"CodingRate4":7,"SyncWord":18,"OutputPower":10,"PreambleLength":8,"CurrentLimit":60.0,"ImplicitHeader":0,"CrcBytes":2}}
10:24:07.732 CMD: lorasend hallo
10:24:07.734 SRC: WebConsole from 192.168.170.45
10:24:07.735 CMD: Grp 0, Cmd 'LORASEND', Idx 1, Len 5, Pld -99, Data 'hallo'
10:24:07.810 RSL: RESULT = {"LoRaSend":"Done"}
```
Board 2:
```
10:24:13.415 CMD: loraconfig
10:24:13.416 SRC: WebConsole from 192.168.170.45
10:24:13.418 CMD: Grp 0, Cmd 'LORACONFIG', Idx 1, Len 0, Pld -99, Data ''
10:24:13.424 RSL: RESULT = {"LoRaConfig":{"Frequency":433.0,"Bandwidth":125,"SpreadingFactor":9,"CodingRate4":7,"SyncWord":18,"OutputPower":10,"PreambleLength":8,"CurrentLimit":60.0,"ImplicitHeader":0,"CrcBytes":2}}
```

**Reply by sfromis** [reply, marked as answer]
With same LoraConfig and same board:
> `{"LoRaReceived":"test\n","RSSI":-61.0,"SNR":8.3}`

And this template, including support for the display:
> `{"NAME":"Lilygo T3_V1.6.1 Lora","GPIO":[1,1,1,1,1,736,1,1,1,1,1,1,1,1,10656,672,1,640,608,10688,1,544,10752,704,6210,0,0,0,1,1,1,1,1,0,0,1],"FLAG":0,"BASE":1}`

**Comment by ChrisVonFrosta** [comment]
Your template did the job! It make it works.
But i do not understand why.
i did the configuration by hand before.

i see no diffence between gpio settings in my setup ("confifure Module") and your template ("configure template").
![grafik](https://github.com/user-attachments/assets/c5acd26a-4d94-4776-a9dc-30e953d25b2a).

> including support for the display
would you share your display.ini?

**Comment by ChrisVonFrosta** [comment]
pls let me ask some addional:
1) The modules has no adressfiltering in Tasmota? do have to parse/filter/implement it my self? ( i know the sync word parameter) 
2) can i make this module to talk with my t3 watch with st126x?

**Reply by sfromis** [reply]
As there is no addressing as such, there can be no filtering, outside of what you can do by screwing up config parameters. That's why there is a clear warning about enabling `LoRaCommand` with no security.

Indeed, if two devices agree on the config and the protocol, they can communicate. That is not merely about what the other thing has of names.

As for display.ini, that's simply the standard SSD1306 display descriptor from
https://github.com/arendst/Tasmota/tree/development/tasmota/displaydesc
Of course, you do need some unused gpio configured with `Option A 3`, and a build including Universal Display.

Maybe what did the trick to get LoRa working was the restart from applying a template? Or whatever....

**Reply by ChrisVonFrosta** [reply]
>That's why there is a clear warning about enabling LoRaCommand with no security.

i thought that only refers to LoraBridgeMode. But it is ok, to do it my self in berry script.
Has Berry some functions for LoRa? Or do i have to have to use a rule for triggering?

>Indeed, if two devices agree on the config and the protocol, they can communicate. That is not merely about what the other thing has of names.

i ask that because i got no success on t-watch with LoraConfig. I saw this [research ](https://blog.classycode.com/lora-sync-word-compatibility-between-sx127x-and-sx126x-460324d1787a)that says it more complicated.

>Maybe what did the trick to get LoRa working was the restart from applying a template?
i don't know, I restored config backup again to test again. Changing config do allways a reboot, doesn't it? as you say, whatever...

... Thx alot!

**Reply by ChrisVonFrosta** [reply]
i can send and receive with the t3_v1.6 (sx127x) on both frequencies.
but the t-watch S3 do not working with.
(all on same LoraConfig)

**Reply by sfromis** [reply]
Lots of features do not really "need" any specific Berry integration, as it works ok with `tasmota.add_rule` and `tasmota.cmd`.

And no, security is also a potential concern without using LoraWan Bridge, within radio range. Depending on how much security you want, Berry does have support for good encryption.

**Reply by arendst** [reply]
> but the t-watch S3 do not working with.

And that one is 433 MHz too? There are different frequency devices on sale.

**Reply by ChrisVonFrosta** [reply]
no. this is nativly 868.
But the t3_v1.6.1 worked on both frequencies.
The different should only be in hardware adjustments to match antenna resonance. So working on the wrong frenquency should only result in a much shorter distance.

the difference seems to be in the syncword configurations. sx126x and sx127x seems to be different more than just the number ;)

i know, it is not ideal for testing but this is what i have here and could test. i would buy a nativly compatible version in future

**Reply by sfromis** [reply]
It may have worked even if you configured with a frequency not supported by the board, but that *may* not mean that it actually used that frequency. And 868 MHz is much more common than 433, meaning that the latter is not great for compatibility with different devices.

**Reply by ChrisVonFrosta** [reply]
i am really sorry and stupid....
all are nativly 433 MHz 
![Unbenannt](https://github.com/user-attachments/assets/e2fd450d-295b-4e80-b8e6-3a90036658f6)

**Reply by nikito7** [reply]
sx126x use a different driver

**Reply by sfromis** [reply]
If that other device also runs Tasmota, that would be a build with `#define USE_LORA_SX126X` and configure the SX126x specific gpios.

**Reply by ChrisVonFrosta** [reply]
> If that other device also runs Tasmota, that would be a build with `#define USE_LORA_SX126X` and configure the SX126x specific gpios.

this was/is already done

**Reply by ChrisVonFrosta** [reply]
both Loratype are detected
